### PR TITLE
Converts DCS to ECS

### PR DESCRIPTION
**Pull request by Cyberboss**
Adding components to datums seems like a little overkill and too complicated. This brings all component stuff up to the /atom level. I can't envision a scenario where we'd need components on datums, but if someone else can, speak now or forever hold your peace.

COMSIG_PARENT_QDELETED now comes from `/atom/Destroy()` instead of `qdel()` because it hasn't been used so far and I don't want to put an istype/typecache shim in there.

### CODE REVIEW

**Comment by XDTM** [comment]
I wanted to add components to viruses or symptoms to let them react to events, would it be possible by adding the component to the infectee?

**Comment by Cyberboss** [comment]
Yeah, because atoms are usually the ones that receive signals. Unless you have a new one planned?

**Comment by Iamgoofball** [comment]
I was going to move GOAP over to being a component and that's a datum and also should be usable on datums

@Cyberboss what is the point of downgrading it to atoms? does it actually do anything beneficial? because having that support IF WE NEED IT is useful.

**Comment by Iamgoofball** [comment]
reagents also need to be moved over to it

same with telecomms, speech, etc.

also what if I want something to exist but not have a physical holder object that uses components?

File: code/datums/components/squeek.dm [hunk 1]
```diff
@@ -24,22 +24,21 @@
 	if(use_delay_override)
 		use_delay = use_delay_override
 
-	if(istype(parent, /atom))
-		RegisterSignal(COMSIG_ATOM_BLOB_ACT, .proc/play_squeak)
-		RegisterSignal(COMSIG_ATOM_HULK_ATTACK, .proc/play_squeak)
```
Comment (Iamgoofball) [inline code/datums/components/squeek.dm]: refactor request, let me pass a list in of COMSIGs so we dont have to copypaste this shit

**Comment by ninjanomnom** [comment]
I agree with goof here. Unless there's a benefit other than the slight Initialize() performance boost, keeping the ability to apply components to datums is worth the cost.

**Comment by optimumtact** [comment]
Yes I am not sure I understand this change

**Pull request closed without merging**